contracts/implementations/nodes/VirtualDevice.sol

```diff
@@ -0,0 +1,235 @@
+//SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.13;
+
+import "../../interfaces/INFT.sol";
+import "../../Eip712/Eip712CheckerInternal.sol";
+import "../../libraries/NodesStorage.sol";
+import "../../libraries/nodes/IntegrationStorage.sol";
+import "../../libraries/nodes/VehicleStorage.sol";
+import "../../libraries/nodes/VirtualDeviceStorage.sol";
+import "../../libraries/MapperStorage.sol";
+
+import "../../shared/Roles.sol";
+import "../../shared/Types.sol";
+
+import "@solidstate/contracts/access/access_control/AccessControlInternal.sol";
+
+/// @title VirtualDevice
+/// @notice Contract that represents the Virtual Device node
+/// @dev It uses the Mapper contract to link Virtual Devices to Vehicles
+contract VirtualDevice is AccessControlInternal {
+    bytes32 private constant MINT_TYPEHASH =
+        keccak256(
+            "MintVirtualDeviceSign(uint256 integrationNode,uint256 vehicleNode,address virtualDeviceAddress)"
```

Comment:
I just thought—we probably can't require the virtual device address here. When the user submits this payload we don't know what the address is.

Reply:
Makes sense. We will still be submitting the addres to the function anwyay, right?

Reply:
Yes!